setup.sh

```diff
@@ -535,16 +535,26 @@ else
 		cat "$wadir/miniserv-conf" >>$cfile
 	fi
 
+	# Test availble hashing formats
+	yescryptpass=`$perl -e 'print crypt("test", "\\$y\\$j9T\\$waHytoaqP/CEnKFroGn0S/\\$fxd5mVc2mBPUc3vv.cpqDckpwrWTyIm2iD4JfnVBi26") eq "\\$y\\$j9T\\$waHytoaqP/CEnKFroGn0S/\\$fxd5mVc2mBPUc3vv.cpqDckpwrWTyIm2iD4JfnVBi26" ? "1\n" : "0\n"'`
+	sha512pass=`$perl -e 'print crypt("test", "\\$6\\$Tk5o/GEE\\$zjvXhYf/dr5M7/jan3pgunkNrAsKmQO9r5O8sr/Cr1hFOLkWmsH4iE9hhqdmHwXd5Pzm4ubBWTEjtMeC.h5qv1") eq "\\$6\\$Tk5o/GEE\\$zjvXhYf/dr5M7/jan3pgunkNrAsKmQO9r5O8sr/Cr1hFOLkWmsH4iE9hhqdmHwXd5Pzm4ubBWTEjtMeC.h5qv1" ? "1\n" : "0\n"'`
 	md5pass=`$perl -e 'print crypt("test", "\\$1\\$A9wB3O18\\$zaZgqrEmb9VNltWTL454R/") eq "\\$1\\$A9wB3O18\\$zaZgqrEmb9VNltWTL454R/" ? "1\n" : "0\n"'`
 
+	salt8=`tr -dc A-Za-z0-9 </dev/urandom | head -c 8 ; echo ''`
+	salt2=`tr -dc A-Za-z0-9 </dev/urandom | head -c 2 ; echo ''`
+
 	ufile=$config_dir/miniserv.users
 	if [ "$crypt" != "" ]; then
 		echo "$login:$crypt:0" > $ufile
 	else
-		if [ "$md5pass" = "1" ]; then
-			$perl -e 'print "$ARGV[0]:",crypt($ARGV[1], "\$1\$XXXXXXXX"),":0\n"' "$login" "$password" > $ufile
+		if [ "$yescryptpass" = "1" ]; then
+			$perl -e 'print "$ARGV[0]:",crypt($ARGV[1], "\$y\$j9T\$'$salt8'"),":0\n"' "$login" "$password" > $ufile
```

Comment:
Pass in $salt8 as a parameter like $login , rather than inserting it into the command.

Reply:
I don't think it makes any difference in this case scenario. If you think it does, could you please show how you expect this to be done?

Reply:
The line would be like : 

```
 $perl -e 'print "$ARGV[0]:",crypt($ARGV[1], "\$y\$j9T\$$ARGV[2]"),":0\n"' "$login" "$password" "$salt8" > $ufile
```

Reply:
Alright, let's do it that way https://github.com/webmin/webmin/commit/ebfb5dbae1606536ac37bd5779180bb263ff78bc https://github.com/webmin/webmin/commit/a21626bd219e4bb68883957b6298dde8ece6cb52.